### ISSUE TITLE
Potential security issue exposing user passwords

### ISSUE THREAD

**Issue by jvandemo** [taktik/freehealth-connector#10]
Currently, some of the endpoints accept a `passPhrase` as a query param in a GET request, such as:
```
https://fhcacc.icure.cloud/sts/token/token-id?ts=123&passPhrase=myPassword123
```

This causes the password `myPassword123 ` to show up as clear text in the URL. Because URL's can be stored in many places (browser history, server logs, network device logs, ...) this may leak the password to external parties.

See https://security.stackexchange.com/questions/142695/is-a-plain-password-in-the-url-a-potential-security-threat for a more in-depth discussion on the impact of using sensitive information in query params.

Although in the context of Oauth, the [Oauth 2.0 Authorization Protocol document on Bearer Tokens](https://tools.ietf.org/html/draft-ietf-oauth-v2-bearer-16#section-4.3) also explains the risk:

> Don't pass bearer tokens in page URLs: Bearer tokens SHOULD NOT be passed in page URLs (for example as query string parameters). Instead, bearer tokens SHOULD be passed in HTTP message headers or message bodies for which confidentiality measures are taken. Browsers, web servers, and other software may not adequately secure URLs in the browser history, web server logs, and other
data structures. If bearer tokens are passed in page URLs, attackers might be able to steal them from the history data, logs, or other unsecured locations.

In the context of Taktik, the bearer token is a password, but the risk is identical.

Given the delicate nature of the health data, putting sensitive data in the query params of a GET request should thus really be avoided.

As a possible solution, an HTTP POST request could be made instead of a GET request.

Are there any plans to support POST requests for all endpoints that require sensitive data? Not having the ability to use a POST request, forces clients to use a GET request, potentially leaking sensitive data.

Thanks in advance!

**Comment by aduchate** [comment]
Though the passwords are not passed in clear during the call on the network. The objections are valid.
I dislike the idea of using the POST request systematically, especially when it is not the adequate verb to be used in the REST logic.

I will add a supported HTTP-HEADER for the tokenId, keystoreId and passPhrase

**Comment by aduchate** [comment]
Fixed in #851964ab789c14a0815b28271c7407645fd1ca23
